### PR TITLE
Make Windows CI faster

### PR DESCRIPTION
**Pull request by liufengyun**
Make Windows CI faster

### CODE REVIEW

**Comment by liufengyun** [comment]
@nicolasstucki It would be good to get this in first so that later PRs will run quickly.

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -102,6 +102,32 @@ jobs:
...
+    if: "github.event_name == 'schedule'
+         || (
+           github.event_name == 'push'
+           && !startsWith(github.event.ref, 'refs/tags/sbt-dotty-')
+         )
+         || (
+           github.event_name == 'pull_request'
+           && !contains(github.event.pull_request.body, '[skip ci]')
+           && !contains(github.event.pull_request.body, '[skip test_windows]')
+           && github.event.pull_request.merged == true
```
Comment (smarter) [inline .github/workflows/ci.yaml:128]: We don't use `github.event.pull_request.merged == true` for any other job, what is its purpose here? As far as I can see we could handle like we handle `test_sbt`: never run on a PR unless `[test_windows_full]` is present in the PR message.

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -102,6 +102,32 @@ jobs:
            && !contains(github.event.pull_request.body, '[skip test_windows]')
```
Comment (smarter) [inline .github/workflows/ci.yaml:102]:
```suggestion
           && !contains(github.event.pull_request.body, '[skip test_windows_fast]')
```

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (smarter) [inline .github/workflows/ci.yaml:128]:
```suggestion
           github.event_name == 'pull_request'
           && !contains(github.event.pull_request.body, '[skip ci]')
           && contains(github.event.pull_request.body, '[test_windows_full]')
```

**Review by smarter — APPROVED** [review]
otherwise lgtm

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (liufengyun) [reply .github/workflows/ci.yaml:128]: It's intended to run `test_windows_full` only on merged PRs. For open PRs, it will run `test_windows_fast`.

Comment (smarter) [reply .github/workflows/ci.yaml:128]: For merged PRs, we get a push event to master so we'll run the test that way.

Comment (liufengyun) [reply .github/workflows/ci.yaml]: Yes, that is simpler.

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (liufengyun) [reply .github/workflows/ci.yaml]: I've removed the line, as now the fast Windows CI finishes in ~8 minutes, usually, there is little need to skip that.

**Comment by nicolasstucki** [comment]
Windows CI finished in `8m 52s` https://github.com/lampepfl/dotty/pull/10553/checks?check_run_id=1473706985

**Review by nicolasstucki — APPROVED** [review]

**Pull request merged**